### DISCUSSION TITLE
useBreakpointValue mock issues, changed with 2.2.5/6

### DISCUSSION THREAD

**Discussion by dbashford** [chakra-ui/chakra-ui#6540] [General] [answered]
### Description

Starting with 2.2.5 my ability to mock useBreakpointValue has become problematic, for now stuck on 2.2.4

### Link to Reproduction

None

### Steps to reproduce

Previous to 2.2.5, I was mocking useBreakpointValue via an import of `@chakra-ui/media-query`. This was the way I found I needed to do it to work. I'm sure there are other ways, but I could not import the hook directly from `@chakra-ui/react`

```js
import * as mediaQueryHooks from '@chakra-ui/media-query';

beforeAll(() => {
  mockUseBreakpoint = jest.spyOn(mediaQueryHooks, 'useBreakpointValue').mockImplementation(() => {
    return true;
  });
});
```

In 2.2.5 that became an issue as the module disappeared

<img width="1288" alt="image" src="https://user-images.githubusercontent.com/581127/183486278-31f8a5bc-5d79-4adc-999a-b87f9908dd6b.png">

In 2.2.6 it seems to have reappeared, but I cannot mock it

<img width="1289" alt="image" src="https://user-images.githubusercontent.com/581127/183486357-124045d5-794a-48ca-9f70-58560e858c87.png">

I suspect there is a better way to be mocking useBreakpointValue that I do not know of, but the way I've been using seems broken.

Any help?

### Chakra UI Version

2.2.5+

### Browser

Unit tests

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by MiroslavPetrik** [comment]


**Reply by dbashford** [reply]
I'm not sure how this is related at all. Are you suggesting that the solution to my issues mocking `useBreakpointValue` is to stop using chakra altogether for solving the problem that `useBreakpointValue` solves?

**Reply by 0ubbe** [reply]
Also blocked with this 😕

**Reply by idrisjaffer89** [reply]
Any update on this issue? Kind of stuck on 2.2.4 version unless this is fixed.

**Reply by veloware** [reply]
Also blocked on this...

**Reply by veloware** [reply]
see below for a possible solution

**Comment by veloware** [comment]
can this be moved back to an issue, definitely feels like something has been broken on version 2.2.5 / 2.2.6 that needs addressed?

**Reply by dbashford** [reply]
@segunadebayo ⬆️

**Comment by veloware** [comment, marked as answer]
following this answer in stackoverflow - https://stackoverflow.com/a/72885576/1849358

you seem to be able to get around this by - 

```
import * as mediaQueryHooks from '@chakra-ui/media-query';

// top of the test file
jest.mock('@chakra-ui/media-query', () => {
  return {
    __esModule: true, //<-- this is important, it doesn't work without it
    ...jest.requireActual('@chakra-ui/media-query'),
  };
});

// then in an individual test...
jest.spyOn(mediaQueryHooks, 'useBreakpointValue').mockImplementationOnce(() => true);
```

that said, it seems like the real solution from Chakra's end, is to make `useBreakpointValue` 'configurable' by setting `configurable: true` in the build process, though not sure of the viability of that from this library's perspective

**Reply by dbashford** [reply]
It has taken me awhile to get back to doing this upgrade, but this worked, thanks!

Quick note for anyone else who may run into this, you can add a permanent across-the-board fix for this.

<img width="284" alt="image" src="https://user-images.githubusercontent.com/581127/235939087-43e3f2a5-7c9f-4a0d-911b-4d390fd14e95.png">

```js
const newMediaQuery = {
  __esModule: true,
  ...jest.requireActual('@chakra-ui/media-query')
};

module.exports = newMediaQuery;
```
